src/createSlice.ts

```diff
@@ -17,8 +17,8 @@ import { createReducer, CaseReducers, CaseReducer } from './createReducer'
 export type SliceActionCreator<P> = PayloadActionCreator<P>
 
 export interface Slice<
-  State = any,
-  ActionCreators extends { [key: string]: any } = { [key: string]: any }
+  CaseReducers extends SliceCaseReducerDefinitions<State, PayloadActions>,
+  State = any
```

Comment:
You could bring this back into the original order by giving the CaseReducers the same default value as ActionCreators had before (that should be working). That way, things might still break for ppl who used this type directly, but maybe not as significantly.

Reply:
Sure, that works.